Sources/TuistScaffold/TemplatesDirectoryLocator.swift

```diff
@@ -0,0 +1,38 @@
+import Basic
+import Foundation
+import TuistCore
+import TuistSupport
+
+public protocol TemplatesDirectoryLocating {
+    /// Returns the path to the custom templates directory if it exists.
+    /// - Parameter at: Path from which we traverse the hierarchy to obtain the templates directory.
+    func locateCustom(at: AbsolutePath) -> AbsolutePath?
```

Comment:
Where the templates are located I think is a good example of a decision we should make on behalf of the developers. It's a subtle thing, but not having to think about where to place the templates:
- Is one less thing that you as a developer have to think about.
- Makes it easier for Tuist to optimize for the convention. If we start allowing customizations, we are opening the door to complexity that we won't be able handle gracefully.

Having said that, I'd remove `locateCustom`.

Reply:
That's what it does - Templates are always at `Tuist/Templates`. The reason it is called `locateCustom` because I plan to add built-in ones, too, which will have a different location

Reply:
Got you. Then I misunderstood what the method is for. What about calling them: userTemplates, tuistTemplates?

Reply:
Sure, I like that 👍